In [22]:
import pandas as pd
import numpy as np

In [23]:
filePath = '../thailand_foods.csv'

In [24]:
df = pd.read_csv(filePath)

In [25]:
df

en_name            th_name  \
0                  Koi               ก้อย   
1            Kuai Chap            กวยจั๊บ   
2            Kuai Tiao         ก๋วยเตี๋ยว   
3      Kuai Tiao Khaek      ก๋วยเตี๋ยวแขก   
4    Kuai Tiao Kua Kai  ก๋วยเตี๋ยวคั่วไก่   
..                 ...                ...   
320              A Kao             อาเกาะ   
321              A Law           อาละหว่า   
322              A Law              อาลัว   
323              O Aew            โอ้เอ๋ว   
324      Ice Cream Pad           ไอติมผัด   

                                          ingredients     course  \
0              Beef+lime juice+fish sauce+chili+herbs  main dish   
1            Pork belly+rice noodles+soy sauce+garlic  main dish   
2    Rice noodles+pork+bean sprouts+garlic+fish sauce  main dish   
3              Beef+noodles+curry powder+coconut milk  main dish   
4                    Chicken+noodles+soy sauce+garlic  main dish   
..                                                ...        ...   
320                                           Unknown    dessert   
321                                           Unknown    dessert   
322                                           Unknown    dessert   
323                                           Unknown    dessert   
324                                           Unknown    dessert   

             province     region  
0    Ubon Ratchathani  Northeast  
1             Bangkok    Central  
2             Bangkok    Central  
3           Ayutthaya    Central  
4             Bangkok    Central  
..                ...        ...  
320           Unknown    Unknown  
321           Unknown    Unknown  
322           Unknown    Unknown  
323           Unknown    Unknown  
324           Unknown    Unknown  

[325 rows x 6 columns]

In [26]:
df.shape

(325, 6)

In [27]:
df.head(10)

en_name            th_name  \
0                      Koi               ก้อย   
1                Kuai Chap            กวยจั๊บ   
2                Kuai Tiao         ก๋วยเตี๋ยว   
3          Kuai Tiao Khaek      ก๋วยเตี๋ยวแขก   
4        Kuai Tiao Kua Kai  ก๋วยเตี๋ยวคั่วไก่   
5         Kuai Tiao Pak Mo  ก๋วยเตี๋ยวปากหม้อ   
6            Kuai Tiao Mon      ก๋วยเตี๋ยวมอญ   
7           Kuai Tiao Ruea     ก๋วยเตี๋ยวเรือ   
8        Kung Chae Nam Pla      กุ้งแช่น้ำปลา   
9  Koey Yeung Kok Siang Bi  เกยย้งก๊กเซียงบี๊   

                                        ingredients     course  \
0            Beef+lime juice+fish sauce+chili+herbs  main dish   
1          Pork belly+rice noodles+soy sauce+garlic  main dish   
2  Rice noodles+pork+bean sprouts+garlic+fish sauce  main dish   
3            Beef+noodles+curry powder+coconut milk  main dish   
4                  Chicken+noodles+soy sauce+garlic  main dish   
5                      Pork+herbs+rice flour+garlic  main dish   
6                Rice noodles+pork+fish sauce+herbs  main dish   
7        Rice noodles+pork+blood broth+bean sprouts  main dish   
8                    Shrimp+fish sauce+garlic+chili  main dish   
9                              Herbs+pork+soy sauce  main dish   

            province     region  
0   Ubon Ratchathani  Northeast  
1            Bangkok    Central  
2            Bangkok    Central  
3          Ayutthaya    Central  
4            Bangkok    Central  
5      Nakhon Pathom    Central  
6         Nonthaburi    Central  
7          Ayutthaya    Central  
8  Coastal provinces      South  
9            Bangkok    Central

In [28]:
df.tail(10)

en_name        th_name                    ingredients   course  \
315        Luk Chup         ลูกชุบ  Mung beans+coconut milk+sugar  dessert   
316        Som Chun         ส้มฉุน         Herbs+fruit+fish sauce    salad   
317   Sankhaya Kuan   สังขยา (กวน)         Flour+coconut milk+egg  dessert   
318  Sankhaya Nueng  สังขยา (นึ่ง)         Flour+coconut milk+egg  dessert   
319          Ang Ku         อังกู๊                        Unknown  dessert   
320           A Kao         อาเกาะ                        Unknown  dessert   
321           A Law       อาละหว่า                        Unknown  dessert   
322           A Law          อาลัว                        Unknown  dessert   
323           O Aew        โอ้เอ๋ว                        Unknown  dessert   
324   Ice Cream Pad       ไอติมผัด                        Unknown  dessert   

    province   region  
315  Various  Various  
316  Various  Various  
317  Various  Various  
318  Various  Various  
319  Unknown  Unknown  
320  Unknown  Unknown  
321  Unknown  Unknown  
322  Unknown  Unknown  
323  Unknown  Unknown  
324  Unknown  Unknown

In [29]:
unknown_count = (df['ingredients'] == 'Unknown').sum()

In [30]:
print("Total rows with Unknown ingredients:", unknown_count)

Total rows with Unknown ingredients: 100


In [35]:
thaiFood = df.drop('th_name', axis=1)

KeyError: "['th_name'] not found in axis"

In [36]:
thaiFood.to_csv("thaiDish.csv", index=False)

In [37]:
df = pd.read_csv('thaiDish.csv')

In [38]:
df

en_name                                       ingredients  \
0                  Koi            Beef+lime juice+fish sauce+chili+herbs   
1            Kuai Chap          Pork belly+rice noodles+soy sauce+garlic   
2            Kuai Tiao  Rice noodles+pork+bean sprouts+garlic+fish sauce   
3      Kuai Tiao Khaek            Beef+noodles+curry powder+coconut milk   
4    Kuai Tiao Kua Kai                  Chicken+noodles+soy sauce+garlic   
..                 ...                                               ...   
320              A Kao                                           Unknown   
321              A Law                                           Unknown   
322              A Law                                           Unknown   
323              O Aew                                           Unknown   
324      Ice Cream Pad                                           Unknown   

        course          province     region  
0    main dish  Ubon Ratchathani  Northeast  
1    main dish           Bangkok    Central  
2    main dish           Bangkok    Central  
3    main dish         Ayutthaya    Central  
4    main dish           Bangkok    Central  
..         ...               ...        ...  
320    dessert           Unknown    Unknown  
321    dessert           Unknown    Unknown  
322    dessert           Unknown    Unknown  
323    dessert           Unknown    Unknown  
324    dessert           Unknown    Unknown  

[325 rows x 5 columns]

In [41]:
mainDishCount = (df['course'] == 'main dish').sum()

In [42]:
print('totalOfMainDish', mainDishCount)

totalOfMainDish 173


In [43]:
dessertCount = (df['course'] == 'dessert').sum()

In [46]:
print('totalDessertCount', dessertCount)

totalDessertCount 70


In [47]:
snackCount = (df['course'] == 'snack').sum()

In [48]:
print('totalSnackCount', snackCount)

totalSnackCount 79


Total Of MainDish = 173,  Total of Dessert = 70, Total of Snack = 79

In [49]:
duplicate_count = df.duplicated().sum()
print("Total duplicate rows:", duplicate_count)

Total duplicate rows: 1


In [50]:
df_no_duplicates = df.drop_duplicates()

In [51]:
df

en_name                                       ingredients  \
0                  Koi            Beef+lime juice+fish sauce+chili+herbs   
1            Kuai Chap          Pork belly+rice noodles+soy sauce+garlic   
2            Kuai Tiao  Rice noodles+pork+bean sprouts+garlic+fish sauce   
3      Kuai Tiao Khaek            Beef+noodles+curry powder+coconut milk   
4    Kuai Tiao Kua Kai                  Chicken+noodles+soy sauce+garlic   
..                 ...                                               ...   
320              A Kao                                           Unknown   
321              A Law                                           Unknown   
322              A Law                                           Unknown   
323              O Aew                                           Unknown   
324      Ice Cream Pad                                           Unknown   

        course          province     region  
0    main dish  Ubon Ratchathani  Northeast  
1    main dish           Bangkok    Central  
2    main dish           Bangkok    Central  
3    main dish         Ayutthaya    Central  
4    main dish           Bangkok    Central  
..         ...               ...        ...  
320    dessert           Unknown    Unknown  
321    dessert           Unknown    Unknown  
322    dessert           Unknown    Unknown  
323    dessert           Unknown    Unknown  
324    dessert           Unknown    Unknown  

[325 rows x 5 columns]

Remove Duplicate Value

In [52]:
duplicates = df[df.duplicated()]
print(duplicates)

    en_name ingredients   course province   region
322   A Law     Unknown  dessert  Unknown  Unknown


In [53]:
df = df.drop(322)

In [54]:
df

en_name                                       ingredients  \
0                  Koi            Beef+lime juice+fish sauce+chili+herbs   
1            Kuai Chap          Pork belly+rice noodles+soy sauce+garlic   
2            Kuai Tiao  Rice noodles+pork+bean sprouts+garlic+fish sauce   
3      Kuai Tiao Khaek            Beef+noodles+curry powder+coconut milk   
4    Kuai Tiao Kua Kai                  Chicken+noodles+soy sauce+garlic   
..                 ...                                               ...   
319             Ang Ku                                           Unknown   
320              A Kao                                           Unknown   
321              A Law                                           Unknown   
323              O Aew                                           Unknown   
324      Ice Cream Pad                                           Unknown   

        course          province     region  
0    main dish  Ubon Ratchathani  Northeast  
1    main dish           Bangkok    Central  
2    main dish           Bangkok    Central  
3    main dish         Ayutthaya    Central  
4    main dish           Bangkok    Central  
..         ...               ...        ...  
319    dessert           Unknown    Unknown  
320    dessert           Unknown    Unknown  
321    dessert           Unknown    Unknown  
323    dessert           Unknown    Unknown  
324    dessert           Unknown    Unknown  

[324 rows x 5 columns]